### ISSUE TITLE
"Cannot read property 'wasThrown' of undefined" when using Console

### ISSUE THREAD

**Issue by faceyspacey** [Microsoft/vscode-chrome-debug#244]
For example if you type `window` you get: "Cannot read property 'wasThrown' of undefined".

As I read in another github issue, it's because it's running in its own process. But you can't even write your own new javascript code like `var foo = 123`. Is the debug console essentially read only? Or am I doing something wrong? 

PS. I'm referring to when you are not currently stepping through code, where it does let you access variables in scope.

**Comment by roblourens** [comment]
No, that just sounds like a bug...

**Comment by roblourens** [comment]
I'll fix the error handling, but that implies that Chrome reported an error when the code was evaluated. Could you set `"verboseDiagnosticLogging": true` and send the logs so I can see what it was? Either upload the file or paste the few lines that are printed when you eval something in the console.

**Comment by faceyspacey** [comment]
```
Verbose logs are written to /Users/jamesgillmore/.vscode/extensions/msjsdiag.debugger-for-chrome-1.0.0/vscode-chrome-debug.txt
OS: darwin x64
Node: v5.10.0
vscode-chrome-debug-core: 2.0.1
debugger-for-chrome: 1.0.0
  ›From client: initialize({"adapterID":"chrome","pathFormat":"path","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true})
  ›To client: {"seq":0,"type":"response","request_seq":1,"command":"initialize","success":true,"body":{"exceptionBreakpointFilters":[{"label":"All Exceptions","filter":"all","default":false},{"label":"Uncaught Exceptions","filter":"uncaught","default":true}],"supportsConfigurationDoneRequest":true,"supportsSetVariable":true,"supportsConditionalBreakpoints":true}}
  ›From client: attach({"name":"Attach","type":"chrome","request":"attach","url":"http://localhost","port":9222,"sourceMaps":true,"webRoot":"/Users/jamesgillmore/React/celebvidy-web","verboseDiagnosticLogging":true})
Discovering targets via http://127.0.0.1:9222/json
  ›Attaching to target: {"description":"","devtoolsFrontendUrl":"/devtools/inspector.html?ws=127.0.0.1:9222/devtools/page/68ed6c81-4f7a-4b89-b97a-6b8cd60f8a84","faviconUrl":"http://localhost/favicon.png","id":"68ed6c81-4f7a-4b89-b97a-6b8cd60f8a84","thumbnailUrl":"/thumb/68ed6c81-4f7a-4b89-b97a-6b8cd60f8a84","title":"React Native for Web","type":"page","url":"http://localhost/","webSocketDebuggerUrl":"ws://127.0.0.1:9222/devtools/page/68ed6c81-4f7a-4b89-b97a-6b8cd60f8a84"}
  ›WebSocket Url: ws://127.0.0.1:9222/devtools/page/68ed6c81-4f7a-4b89-b97a-6b8cd60f8a84
  ›To target: {"id":1,"method":"Runtime.enable"}
  ›To target: {"id":2,"method":"Debugger.enable"}
  ›To target: {"id":3,"method":"Console.enable"}
  ›To target: {"id":4,"method":"Runtime.runIfWaitingForDebugger"}
  ›To target: {"id":5,"method":"Runtime.run"}
  ›From target: {"method":"Runtime.executionContextCreated","params":{"context":{"id":40,"isDefault":false,"origin":"chrome-extension://bhlhnicpbhignbdhedgjhgdocnmhomnp","name":"ColorZilla","frameId":"18297.1"}}}
  ›From target: {"method":"Runtime.executionContextCreated","params":{"context":{"id":39,"isDefault":false,"origin":"chrome-extension://lmhkpmbekcpmknklioeibfkpmmfibljd","name":"Redux DevTools","frameId":"18297.1"}}}
  ›From target: {"method":"Runtime.executionContextCreated","params":{"context":{"id":38,"isDefault":false,"origin":"chrome-extension://fmkadmapgofadopljbjfkapdkoienihi","name":"React Developer Tools","frameId":"18297.1"}}}
  ›From target: {"method":"Runtime.executionContextCreated","params":{"context":{"id":37,"isDefault":true,"origin":"http://localhost","name":"","frameId":"18297.1"}}}
  ›From target: {"id":1,"result":{}}
›From target: {"method":"Debugger.scriptParsed","params":{"scriptId":"1062","url":"http://localhost/static/bundle.js","startLine":0,"startColumn":0,"endLine":71649,"endColumn":28100412,"executionContextId":37,"hash":"58BE45B0AFFAA615AB86425539CF7070126A33A0","isContentScript":false,"isInternalScript":false,"isLiveEdit":false,"sourceMapURL":"data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly8vd2VicGFjay9ib290c3RyYXAgMTcwZTQyOTFhZTg2OWE0MTVhNDgiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9yZWFjdC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL2ludmFyaWFudC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL3dhcm5pbmcuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvcmVhY3RQcm9kSW52YXJpYW50LmpzIiwid2VicGFjazovLy8uL34vcmVhY3QtbmF0aXZlLXdlYi9kaXN0L2luZGV4LmpzIiwid2VicGFjazovLy8uL34vb2JqZWN0LWFzc2lnbi9pbmRleC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL0V4ZWN1dGlvbkVudmlyb25tZW50LmpzIiwid2VicGFjazovLy8uL34vaW52YXJpYW50L2Jyb3dzZXIuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvUmVhY3RET01Db21wb25lbnRUcmVlLmpzIiwid2VicGFjazovLy8uL34vcmVhY3QtbmF0aXZlLXdlYi9kaXN0L2FwaXMvU3R5bGVTaGVldC9pbmRleC5qcyIsIndlYnBhY2s6Ly8vLi9+L3JlYWN0LXJvdXRlci9saWIvcm91dGVyV2FybmluZy5qcyIsIndlYnBhY2s6Ly8vLi9+L3JlYWN0LW5hdGl2ZS13ZWIvZGlzdC9jb21wb25lbnRzL1ZpZXcvaW5kZXguanMiLCJ3ZWJwYWNrOi8vLy4vfi9hbmltYXRlZC9saWIvQW5pbWF0ZWQuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvUmVhY3RJbnN0cnVtZW50YXRpb24uanMiLCJ3ZWJwYWNrOi8vLy4vYXBwL2ltYWdlcy9sb2dvLXBsYXktYnV0dG9uLnBuZyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL2VtcHR5RnVuY3Rpb24uanMiLCJ3ZWJwYWNrOi8vLy4[...]
Paths.scriptParsed: could not resolve http://localhost/static/bundle.js to a file under webRoot: /Users/jamesgillmore/React/celebvidy-web. It may be external or served directly from the server's memory (and that's OK).
SourceMaps.getMapForGeneratedPath: Finding SourceMap for http://localhost/static/bundle.js by URI: data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly8vd2VicGFjay9ib290c3RyYXAgMTcwZTQyOTFhZTg2OWE0MTVhNDgiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9yZWFjdC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL2ludmFyaWFudC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL3dhcm5pbmcuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvcmVhY3RQcm9kSW52YXJpYW50LmpzIiwid2VicGFjazovLy8uL34vcmVhY3QtbmF0aXZlLXdlYi9kaXN0L2luZGV4LmpzIiwid2VicGFjazovLy8uL34vb2JqZWN0LWFzc2lnbi9pbmRleC5qcyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL0V4ZWN1dGlvbkVudmlyb25tZW50LmpzIiwid2VicGFjazovLy8uL34vaW52YXJpYW50L2Jyb3dzZXIuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvUmVhY3RET01Db21wb25lbnRUcmVlLmpzIiwid2VicGFjazovLy8uL34vcmVhY3QtbmF0aXZlLXdlYi9kaXN0L2FwaXMvU3R5bGVTaGVldC9pbmRleC5qcyIsIndlYnBhY2s6Ly8vLi9+L3JlYWN0LXJvdXRlci9saWIvcm91dGVyV2FybmluZy5qcyIsIndlYnBhY2s6Ly8vLi9+L3JlYWN0LW5hdGl2ZS13ZWIvZGlzdC9jb21wb25lbnRzL1ZpZXcvaW5kZXguanMiLCJ3ZWJwYWNrOi8vLy4vfi9hbmltYXRlZC9saWIvQW5pbWF0ZWQuanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvUmVhY3RJbnN0cnVtZW50YXRpb24uanMiLCJ3ZWJwYWNrOi8vLy4vYXBwL2ltYWdlcy9sb2dvLXBsYXktYnV0dG9uLnBuZyIsIndlYnBhY2s6Ly8vLi9+L2ZianMvbGliL2VtcHR5RnVuY3Rpb24uanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC1yZWR1eC9saWIvaW5kZXguanMiLCJ3ZWJwYWNrOi8vLy4vfi9yZWFjdC9saWIvUmVhY3RFbGVtZW50LmpzIiwid2VicGFjazovLy8uL34vcmVhY3QtZG9tL2luZGV4LmpzIiwid2VicGFjazovLy8uL34vcmVhY3QvbGliL0V2ZW50Q29uc3RhbnRzLmpzIiwid2VicGFjazovLy8uL34vcmVhY3QvbGliL1JlYW[...]
SourceMaps.getMapForGeneratedPath: Using inlined sourcemap in http://localhost/static/bundle.js
SourceMap: creating for http://localhost/static/bundle.js
SourceMap: sourceRoot: 
SourceMap: sources: ["webpack:///webpack/bootstrap 170e4291ae869a415a48","webpack:///./~/react/react.js","webpack:///./~/fbjs/lib/invariant.js","webpack:///./~/fbjs/lib/warning.js","webpack:///./~/react/lib/reactProdInvariant.js","webpack:///./~/react-native-web/dist/index.js","webpack:///./~/object-assign/index.js","webpack:///./~/fbjs/lib/ExecutionEnvironment.js","webpack:///./~/invariant/browser.js","webpack:///./~/react/lib/ReactDOMComponentTree.js","webpack:///./~/react-native-web/dist/apis/StyleSheet/index.js","webpack:///./~/react-router/lib/routerWarning.js","webpack:///./~/react-native-web/dist/components/View/index.js","webpack:///./~/animated/lib/Animated.js","webpack:///./~/react/lib/ReactInstrumentation.js","webpack:///./app/images/logo-play-button.png","webpack:///./~/fbjs/lib/emptyFunction.js","webpack:///./~/react-redux/lib/index.js","webpack:///./~/react/lib/ReactElement.js","webpack:///./~/react-dom/index.js","webpack:///./~/react/lib/EventConstants.js","webpack:///./~/react/lib/ReactComponentTreeHook.js","webpack:///./~/react/lib/ReactUpdates.js","webpack:///(webpack)/buildin/global.js","webpack:///./~/fbjs/lib/keyOf.js","webpack:///./~/history/~/warning/browser.js","webpack:///./~/react/lib/PooledClass.js","webpack:///./~/react/lib/SyntheticEvent.js","webpack:///./~/animated/lib/AnimatedWithChildren.js","webpack:///./~/lodash/_root.js","webpack:///./~/react/lib/ReactCurrentOwner.js","webpack:///./app/widgets/Button.js","webpack:///./app/widgets/Modal.js","[...]
SourceMap: webRoot: /Users/jamesgillmore/React/celebvidy-web
SourceMap: no sourceRoot specified, using webRoot + script path dirname: /Users/jamesgillmore/React/celebvidy-web/static
SourceMap: mapping webpack:///webpack/bootstrap 170e4291ae869a415a48 => /Users/jamesgillmore/React/celebvidy-web/webpack/bootstrap 170e4291ae869a415a48, via sourceMapPathOverrides entry - "webpack:///*": "/Users/jamesgillmore/React/celebvidy-web/*"
```

**A BUNCH OF WEBPACK SOURCE MAP INFO WHICH LOOKS CORRECT -- I'D PASTE IT BUT IT'S EXTREMELY LONG AND THERE IS NO OPTION TO SELECT ALL OUTPUT IN THE CONSOLE**

```
  ›From target: {"method":"Console.messageAdded","params":{"message":{"source":"console-api","level":"log","text":"Loaded state:","type":"log","url":"http://localhost/static/bundle.js","line":35011,"column":20,"parameters":[{"type":"string","value":"Loaded state:"},{"type":"object","className":"Object","description":"Object","objectId":"{\"injectedScriptId\":37,\"id\":1}"}],"stack":{"callFrames":[{"functionName":"","scriptId":"1062","url":"http://localhost/static/bundle.js","lineNumber":35011,"columnNumber":20}]},"timestamp":1475794173.82871,"executionContextId":37}}}
Loaded state: Object
  ›From target: {"method":"Console.messageAdded","params":{"message":{"source":"console-api","level":"log","text":"[HMR] connected","type":"log","url":"http://localhost/static/bundle.js","line":28470,"column":30,"parameters":[{"type":"string","value":"[HMR] connected"}],"stack":{"callFrames":[{"functionName":"handleOnline","scriptId":"1062","url":"http://localhost/static/bundle.js","lineNumber":28470,"columnNumber":30}]},"timestamp":1475794176.20689,"executionContextId":37}}}
[HMR] connected
  ›From target: {"id":3,"result":{}}
  ›From target: {"error":{"code":-32601,"message":"'Runtime.runIfWaitingForDebugger' wasn't found"},"id":4}
  ›From target: {"id":5,"result":{}}
  ›To client: {"seq":0,"type":"event","event":"initialized"}
  ›To client: {"seq":0,"type":"response","request_seq":2,"command":"attach","success":true}
  ›From client: setExceptionBreakpoints({"filters":["uncaught"]})
  ›To target: {"id":6,"method":"Debugger.setPauseOnExceptions","params":{"state":"uncaught"}}
  ›From target: {"id":6,"result":{}}
  ›To client: {"seq":0,"type":"response","request_seq":3,"command":"setExceptionBreakpoints","success":true}
  ›From client: configurationDone(undefined)
  ›To client: {"seq":0,"type":"response","request_seq":4,"command":"configurationDone","success":true}
```

**AND THEN A FEW SECONDS THIS USUALLY APPEARS:**

```
 ›From client: completions({"text":"G","column":2,"line":1})
Error: unknowncommand
    at Object.errP (/Users/jamesgillmore/.vscode/extensions/msjsdiag.debugger-for-chrome-1.0.0/node_modules/vscode-chrome-debug-core/out/src/utils.js:195:13)
    at dispatchRequest (/Users/jamesgillmore/.vscode/extensions/msjsdiag.debugger-for-chrome-1.0.0/node_modules/vscode-chrome-debug-core/out/src/chrome/chromeDebugSession.js:117:23)
    at ProtocolServer._handleData (/Users/jamesgillmore/.vscode/extensions/msjsdiag.debugger-for-chrome-1.0.0/node_modules/vscode-chrome-debug-core/node_modules/vscode-debugadapter/lib/protocol.js:104:38)
    at Socket.<anonymous> (/Users/jamesgillmore/.vscode/extensions/msjsdiag.debugger-for-chrome-1.0.0/node_modules/vscode-chrome-debug-core/node_modules/vscode-debugadapter/lib/protocol.js:24:60)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at readableAddChunk (_stream_readable.js:153:18)
    at Socket.Readable.push (_stream_readable.js:111:10)
    at Pipe.onread (net.js:531:20)
  ›To client: {"seq":0,"type":"response","request_seq":5,"command":"completions","success":false,"message":"[debugger-for-chrome] unknowncommand"}
  ›From client: completions({"text":"w","column":2,"line":1})
```


**AND IF I TRY TO DO ANYTHING IN THE CONSOLE:**

```
window.foo
Cannot read property 'wasThrown' of undefined
  ›From client: evaluate({"expression":"window.foo","context":"repl"})
  ›To target: {"id":7,"method":"Runtime.evaluate","params":{"expression":"window.foo","contextId":1,"returnByValue":false,"silent":true}}
  ›From target: {"error":{"code":-32000,"message":"Cannot find context with specified id"},"id":7}
  ›To client: {"seq":0,"type":"response","request_seq":8,"command":"evaluate","success":false,"message":"Cannot read property 'wasThrown' of undefined"}
```


I'd say I have a pretty standard webpack setup, minus the fact that I'm using the sourcemap plugin as previously mentioned. So this should be helpful to the many webpack users.


Here's my webpack config:

```
const path = require('path')
const webpack = require('webpack')
const ExtendedDefinePlugin = require('extended-define-webpack-plugin');

module.exports = {
  devServer: {
    contentBase: path.join(__dirname, 'src')
  },
  //devtool: 'eval',
  //devtool: '#cheap-module-eval-source-map',
  entry: [
    'react-hot-loader/patch',
    'webpack-hot-middleware/client?path=http://localhost:80/__webpack_hmr&timeout=20000&reload=true&quiet=false&noInfo=false',
    //'webpack-dev-server/client?http://localhost:3000',
    //'webpack/hot/only-dev-server',
    path.join(__dirname, '../index.web.js')
  ],
  output: {
    path: path.join(__dirname, 'build'),
    filename: 'bundle.js',
    publicPath: '/static/'
  },
  module: {
    loaders: [
      {
        test: /\.css$/,
        loader: "style-loader!css-loader"
      },
      {
        test: /\.js$/,
        //include: path.join(__dirname, '../app'),
        //exclude: /node_modules\/(?!(react-native-vector-icons)\/).*/,
        exclude: /node_modules\/(?!(react-native-animatable)\/).*/,
        loader: 'babel',
        query: {
          "presets": ["es2015", "react", "stage-0"],
          "plugins": ["react-hot-loader/babel", "transform-decorators-legacy"],
          cacheDirectory: true,
        },
      },
      {
        test: /\.(gif|jpe?g|png|svg)$/,
        loader: 'url-loader',
        query: { name: '[name].[hash:16].[ext]' }
      }
    ]
  },
  plugins: [
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development')
    }),
    new ExtendedDefinePlugin({
      CONFIG: {
        development: true,
        production: false,
      },
      __DEV__: true
    }),
    //new webpack.optimize.DedupePlugin(),
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoErrorsPlugin(),
    new webpack.SourceMapDevToolPlugin({
      test: /\.js$/, //the damn thing breaks down and source map line numbers dont match if u get too fancy to optimize performance
      exclude: /node_modules/,

      //module: false,
      //columns: false,
      //lineToLine: false,
    }),
  ],
  devtool: false,
  resolve: {
    alias: {
      'react-native': 'react-native-web'
    }
  }
}

```

**Comment by roblourens** [comment]
Dangit, I know what the problem was. It was caused by my workaround for a node bug - https://github.com/nodejs/node/issues/8426. I suspect that it would work if you only have one tab open in Chrome, or something, not totally sure what determines those IDs. And it should still work when you're at a breakpoint.

The 'unknowncommand' error is ok, it's caused by a bug in vscode which will be fixed in 1.6. But shouldn't cause a problem.

**Comment by roblourens** [comment]
Alright should be fixed in 1.2.0
